File: frontend/src/microservices/tour-group/objects/UploadTourGroupCoverImageResponse.ts
// 本文件定义 tour-group 模块的 `UploadTourGroupCoverImageResponse`，作为封面上传响应并提供 JSON 编解码。

export type UploadTourGroupCoverImageResponse = {
  assetId: string
  publicUrl: string
  originalFileName: string
  mimeType: string
  fileSize: number
}

export const uploadTourGroupCoverImageResponseFromJson = (json: string): UploadTourGroupCoverImageResponse =>
  JSON.parse(json) as UploadTourGroupCoverImageResponse

export const uploadTourGroupCoverImageResponseToJson = (value: UploadTourGroupCoverImageResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/UploadTourGroupCoverImageResponse.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadTourGroupCoverImageResponse(
    assetId: String,
    publicUrl: String,
    originalFileName: String,
    mimeType: String,
    fileSize: Long
)
object UploadTourGroupCoverImageResponse:
  given sourceEncoder: Encoder[UploadTourGroupCoverImageResponse] = deriveEncoder
  given sourceDecoder: Decoder[UploadTourGroupCoverImageResponse] = deriveDecoder
